molgenis-data/src/main/java/org/molgenis/data/support/DefaultEntity.java

```diff
@@ -275,6 +272,16 @@ public Entity getEntity(String attributeName)
 	@Override
 	public Iterable<Entity> getEntities(String attributeName)
 	{
+		AttributeMetaData attribute = entityMetaData.getAttribute(attributeName);
+		if (attribute == null) throw new UnknownAttributeException(attributeName);
+
+		FieldType dataType = attribute.getDataType();
+		if (!(dataType instanceof MrefField) && !(dataType instanceof XrefField))
```

Comment:
This will fail for other implementations of Entity. Do you know a use case where DefaultEntity contains a Iterable<Entity> for a CATEGORICAL_XREF/XREF/FILE attribute?